cli/crates/cli/src/start/configuration.rs

```diff
@@ -0,0 +1,514 @@
+            expose_headers = "any"
+        "#};
+
+        let config: Config = toml::from_str(input).unwrap();
+        let cors = config.cors.unwrap();
+
+        assert_eq!(Some(AnyOrAsciiStringArray::Any), cors.expose_headers);
+    }
+
+    #[test]
+    fn cors_expose_headers_explicit() {
+        let input = indoc! {r#"
+            [cors]
+            expose_headers = ["Content-Type"]
+        "#};
+
+        let config: Config = toml::from_str(input).unwrap();
+        let cors = config.cors.unwrap();
+
+        let expected = AnyOrAsciiStringArray::Explicit(vec![AsciiString::from_ascii(b"Content-Type").unwrap()]);
+
+        assert_eq!(Some(expected), cors.expose_headers);
+    }
+
+    #[test]
+    fn cors_expose_headers_invalid() {
+        let input = indoc! {r#"
+            [cors]
+            expose_headers = ["😂😂😂"]
+        "#};
+
+        let error = toml::from_str::<Config>(input).unwrap_err();
+
+        insta::assert_snapshot!(&error.to_string(), @r###"
+        TOML parse error at line 2, column 18
+          |
+        2 | expose_headers = ["😂😂😂"]
+          |                  ^^^^^^^^^^^^^^^^
+        data did not match any variant of untagged enum AnyOrAsciiStringArray
+        "###);
+    }
+
+    #[test]
+    fn cors_allow_private_network_default() {
+        let input = indoc! {r#"
+            [cors]
+        "#};
+
+        let config: Config = toml::from_str(input).unwrap();
+        let cors = config.cors.unwrap();
+
+        assert!(!cors.allow_private_network);
+    }
+
+    #[test]
+    fn cors_allow_private_network_explicit() {
+        let input = indoc! {r#"
+            [cors]
+            allow_private_network = true
+        "#};
+
+        let config: Config = toml::from_str(input).unwrap();
+        let cors = config.cors.unwrap();
+
+        assert!(cors.allow_private_network);
+    }
+
+    #[test]
+    fn operation_limits() {
+        let input = indoc! {r#"
+            [operation_limits]
+            depth = 3
+            height = 10
+            aliases = 100
+            root_fields = 10
+            complexity = 1000
+        "#};
+
+        let config: Config = toml::from_str(input).unwrap();
+        let operation_limits = config.operation_limits.unwrap();
+
+        let expected = OperationLimitsConfig {
+            depth: 3,
+            height: 10,
+            aliases: 100,
+            root_fields: 10,
+            complexity: 1000,
+        };
+
+        assert_eq!(expected, operation_limits);
+    }
+
+    #[test]
+    fn operation_limits_with_missing_values() {
+        let input = indoc! {r#"
+            [operation_limits]
+            depth = 3
+            height = 10
+        "#};
+
+        let error = toml::from_str::<Config>(input).unwrap_err();
+
+        insta::assert_snapshot!(&error.to_string(), @r###"
+        TOML parse error at line 1, column 1
+          |
+        1 | [operation_limits]
+          | ^^^^^^^^^^^^^^^^^^
+        missing field `aliases`
```

Comment:
Nitpick, but could we limit depth and height without limiting aliases? That seems to make sense